src/main/java/com/productdock/library/search/kafka/consumer/KafkaConsumer.java

```diff
@@ -36,4 +33,10 @@ public synchronized void listen(BookAvailabilityMessage bookAvailabilityMessage)
     public synchronized void listen(BookRatingMessage bookRatingMessage) {
         bookService.updateBookRating(bookRatingMessage);
     }
+
+    @KafkaListener(topics = "${spring.kafka.topic.book-recommendation}",
+            containerFactory = "bookRecommendedMessageKafkaListenerContainerFactory")
+    public synchronized void listen(BookRecommendedMessage bookRecommendedMessage) {
+        bookService.updateBookRecommendations(bookRecommendedMessage);
```

Comment:
It's better for method name to be "recommendBook". As parameter pass bookId instead of whole message